src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -451,7 +487,7 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 
 	handleCommandFinished(exitCode: number | undefined, options?: IHandleCommandOptions): void {
 		if (this._isWindowsPty) {
```

Comment:
do we need to be worried about `handleCommandExecuted` happening at the same time as `handleCommandFinished` given they both could be awaiting the same promise?

Reply:
Yes let's make sure the task has run before handling finished as it looks like `handleCommandFinished` does some validation later on (`if (this._currentCommand.commandStartMarker === undefined || !this._terminal.buffer.active) {`). It would also be nicer to the awaiting inside `_preHandleCommandFinishedWindows` and make that `async`